File: xfactorial.py
def factorial(n):
	if n == 1:
		return 1
	else:
		return n*factorial(n-1)

def xfactorial(m):
	if m == 0:
		return 1
	elif m == 1:
		return m
	elif m < 0:
		return None
	else:
		return factorial(m)

print("################ xfactorial(m) ##################")
a = 5
print(xfactorial(a))

	

File: xfactorial.go
package main

func factorial(n float64, stop float64) float64{
	if n == stop{
		return n
	} else{
		if n == 0.0{
			return 1.0
		} else{
			return n * factorial(n-1.0, stop);
		}
	}
}

func xfactorial(channel chan float64, start float64, end float64){
	// Reverse order, calculate higher value first. Start value is stop condition for recursive function factorial(x)
	channel <- factorial(end, start);
}